# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../udev/hotfix.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fix build w/ dietlibc, probably related to implicit glibc include hierarchy:

extras/ata_id/ata_id.o: In function `disk_identify_fixup_uint16':
ata_id.c:(.text+0xd1): undefined reference to `le16toh'

  - Rene Rebe <rene@exactcode.de>

--- udev-147/extras/ata_id/ata_id.c.vanilla	2011-01-05 19:15:42.000000000 +0100
+++ udev-147/extras/ata_id/ata_id.c	2011-01-05 19:15:53.000000000 +0100
@@ -39,6 +39,7 @@
 #include <linux/hdreg.h>
 #include <linux/fs.h>
 #include <arpa/inet.h>
+#include <endian.h>
 
 #include "libudev.h"
 #include "libudev-private.h"
--- udev-167/libudev/libudev-device-private.c.vanilla	2018-08-08 17:46:08.247860535 +0000
+++ udev-167/libudev/libudev-device-private.c	2018-08-08 17:46:34.963860133 +0000
@@ -18,6 +18,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #include "libudev.h"
 #include "libudev-private.h"
--- udev-167/libudev/libudev-private.h.vanilla	2018-08-08 17:48:01.711858828 +0000
+++ udev-167/libudev/libudev-private.h	2018-08-08 17:48:06.659858754 +0000
@@ -16,6 +16,8 @@
 #include <signal.h>
 #include <stdint.h>
 #include <stdbool.h>
+#include <sys/sysmacros.h>
+
 #include "libudev.h"
 
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
